pymc/pytensorf.py

```diff
@@ -141,11 +157,18 @@ def convert_data(data) -> np.ndarray | Variable:
     # Otherwise we only convert the precision.
     return smarttypeX(ret)
 
-
-@_as_tensor_variable.register(pd.Series)
-@_as_tensor_variable.register(pd.DataFrame)
-def dataframe_to_tensor_variable(df: pd.DataFrame, *args, **kwargs) -> TensorVariable:
-    return pt.as_tensor_variable(df.to_numpy(), *args, **kwargs)
+if pl is not None:
+    @_as_tensor_variable.register(pd.Series)
+    @_as_tensor_variable.register(pd.DataFrame)
+    @_as_tensor_variable.register(pl.DataFrame)
+    @_as_tensor_variable.register(pl.Series)
+    def dataframe_to_tensor_variable(df: pd.DataFrame | pl.DataFrame, *args, **kwargs) -> TensorVariable:
+        return pt.as_tensor_variable(df.to_numpy(), *args, **kwargs)
+else:
+    @_as_tensor_variable.register(pd.Series)
+    @_as_tensor_variable.register(pd.DataFrame)
+    def dataframe_to_tensor_variable(df: pd.DataFrame, *args, **kwargs) -> TensorVariable:
+        return pt.as_tensor_variable(df.to_numpy(), *args, **kwargs)
```

Comment:
This is more succinct. Also type hint of df was wrong, so I just removed it.

```suggestion
@_as_tensor_variable.register(pd.Series)
@_as_tensor_variable.register(pd.DataFrame)
def dataframe_to_tensor_variable(df, *args, **kwargs) -> TensorVariable:
    return pt.as_tensor_variable(df.to_numpy(), *args, **kwargs)

if pl is not None:
    @_as_tensor_variable.register(pl.DataFrame)
    @_as_tensor_variable.register(pl.Series)
    def polars_dataframe_to_tensor_variable(df, *args, **kwargs) -> TensorVariable:
        return pt.as_tensor_variable(df.to_numpy(), *args, **kwargs)

```